src/torchpme/tuning/ewald.py

```diff
@@ -0,0 +1,196 @@
+        is supported
+    :param neighbor_indices: torch.Tensor with the ``i,j`` indices of neighbors for
+        which the potential should be computed in real space.
+    :param neighbor_distances: torch.Tensor with the pair distances of the neighbors for
+        which the potential should be computed in real space.
+    :param ns_lo: Minimum number of spatial resolution along each axis
+    :param ns_hi: Maximum number of spatial resolution along each axis
+    :param accuracy: Recomended values for a balance between the accuracy and speed is
+        :math:`10^{-3}`. For more accurate results, use :math:`10^{-6}`.
+
+    :return: Tuple containing a float of the optimal smearing for the :class:
+        `CoulombPotential`, and a dictionary with the parameters for
+        :class:`EwaldCalculator`.
+
+    Example
+    -------
+    >>> import torch
+    >>> positions = torch.tensor(
+    ...     [[0.0, 0.0, 0.0], [0.4, 0.4, 0.4]], dtype=torch.float64
+    ... )
+    >>> charges = torch.tensor([[1.0], [-1.0]], dtype=torch.float64)
+    >>> cell = torch.eye(3, dtype=torch.float64)
+    >>> smearing, parameter = tune_ewald(
+    ...     charges, cell, positions, cutoff=4.4, accuracy=1e-1
+    ... )
+
+    """
+    _validate_parameters(charges, cell, positions, exponent)
+    min_dimension = float(torch.min(torch.linalg.norm(cell, dim=1)))
+    params = [{"lr_wavelength": min_dimension / ns} for ns in range(ns_lo, ns_hi + 1)]
+    tuner = GridSearchTuner(
+        charges,
+        cell,
+        positions,
+        cutoff,
+        exponent=exponent,
+        neighbor_indices=neighbor_indices,
+        neighbor_distances=neighbor_distances,
+        calculator=EwaldCalculator,
+        error_bounds=EwaldErrorBounds(charges=charges, cell=cell, positions=positions),
+        params=params,
+    )
+    smearing = tuner.estimate_smearing(accuracy)
+    errs, timings = tuner.tune(accuracy)
+
+    if any(err < accuracy for err in errs):
+        # There are multiple errors below the accuracy, return the one with the shortest
+        # calculation time. The timing of those parameters leading to an higher error
+        # than the accuracy are set to infinity
+        return smearing, params[timings.index(min(timings))]
+    # No parameter meets the requirement, return the one with the smallest error
+    return smearing, params[errs.index(min(errs))]
+
+
+class EwaldErrorBounds(TuningErrorBounds):
+    r"""
+    Error bounds for :class:`torchpme.calculators.ewald.EwaldCalculator`.
+
+    The error formulas are given `online
+    <https://www2.icp.uni-stuttgart.de/~icp/mediawiki/images/4/4d/Script_Longrange_Interactions.pdf>`_
+    (now not available, need to be updated later). Note the difference notation between
+    the parameters in the reference and ours:
+
+    .. math::
+
+        \alpha &= \left( \sqrt{2}\,\mathrm{smearing} \right)^{-1}
+
+        K &= \frac{2 \pi}{\mathrm{lr\_wavelength}}
+
+        r_c &= \mathrm{cutoff}
```

Comment:
This docstring seems to be very similar to the tuning function.

Maybe write someting that this class implemenents the error bounds for the real and FT part of the ewald summation ...